### PR TITLE
Formalize, document and automate release process for AM controller docker image

### PR DESCRIPTION
**Pull request by davemasselink**
Fixes issue #37

## Proposed Changes
  - Added and enhanced release process via Makefile and travis yaml config
  - Updated documentation related to process
  
## Testing Done
  - all commits trigger hash tagged artifact creation on dockerhub
  - all tags trigger tag named tagged artifact creation on dockerhub
https://hub.docker.com/repository/registry-1.docker.io/keikoproj/active-monitor/tags

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
-# Image URL to use all building/pushing image targets
+# Image URL to use all building/pushing image targets (IMG will be overwritten during CI operations)
 IMG ?= keikoproj/active-monitor:latest
+LATEST_IMG ?= keikoproj/active-monitor:latest
```
Comment (davemasselink) [inline Makefile]: the addition of this envVar such that the `latest` tag will always be built/published makes sense to me... but make sure you agree. If we don't want to tag as `latest` on git tag triggered builds, we'd want to change this back

Comment (mattouille) [reply Makefile]: If `latest` is essentially pointing to tip then I think it makes sense.

Comment (davemasselink) [reply Makefile]: ok, thanks for your thoughts

I'm treating `latest` here as the latest built version, regardless of where it came from. Essentially, it means `latest` will be bleeding edge and changing frequently.

as always, the users of AM will need to reference a version tagged docker image (or maybe git hash tagged) for consistency. `latest` won't, and isn't meant to, provide that

**Comment by davemasselink** [comment]
> LGTM!
> Release changes work fine with Kustomize 2.1.0

thanks for (initially unknowingly) testing the project with >v3.1 Kustomize... good to know we'll eventually need to update the makefile's deploy target at some point in the future

**Pull request merged**